drivers/ethernet/eth_adin2111.c

```diff
@@ -380,7 +391,7 @@ static int eth_adin2111_send_oa_frame(const struct device *dev, struct net_pkt *
 		}
 		if (i == chunks) {
 			hdr |= ADIN2111_OA_DATA_HDR_EV;
-			hdr |= (ctx->oa_cps - 1) << ADIN2111_OA_DATA_HDR_EBO;
+			hdr |= ebo << ADIN2111_OA_DATA_HDR_EBO;
```

Comment:
When at the last chunk, you should consider adding `ebo` to `cur` instead of `cur += ctx->oa_cps;` (line 397). This would ensure that the frame length passed to the SPI layer is exactly correct, although I don't think this is a functional issue.